influence_benchmark/RL/base_iteration.py

```diff
@@ -146,39 +144,54 @@ def _save_kwargs(self, kwargs):
         with open(str(self.traj_dir / "kwargs.yaml"), "w+") as outfile:
             yaml.dump(self.kwargs_to_save, outfile, default_flow_style=False)
 
+    def setup_backends(self, agent_device, env_device, lora_path=None):
+        backends = {}
+        if agent_device != env_device or self.inference_quantization is not None:
+            # Assuming that if this is the case, we can't share any backend at all. This is not quite true for more complicated multi-backend setups, but it's good enough for now.
+            for model_type, model_name in self.model_names.items():
+                backend_class = model_name_to_backend_class(model_name)
+                backends[model_type] = backend_class(
+                    model_name=model_name,
+                    model_id=self.agent_model_id,  # type: ignore
+                    device=agent_device if model_type == "agent" else env_device,
+                    lora_path=lora_path,
+                    max_tokens_per_minute=self.max_tokens_per_minute,
+                    inference_quantization=self.inference_quantization if model_type == "agent" else None,
```

Comment:
If we were to use a  bigger model for the env we could potentially want quantization, but I agree the most likely scenario is we want quant on the agent and no quant on the env